src/core/MIGRATION_EXAMPLES.md

```diff
@@ -0,0 +1,255 @@
+config service due to the way that config is tied to the `kibana.json` file
+(which does not exist for legacy plugins).
+
+There is a workaround though:
+- Create a New Platform plugin that contains your plugin's config schema in the new format
+- Expose the config from the New Platform plugin in its setup contract
+- Read the config from the setup contract in your legacy plugin
+
+#### Create a New Platform plugin
+
+For example, if wanted to move the legacy `timelion` plugin's configuration to
+the New Platform, we could create a NP plugin with the same name in
+`src/plugins/timelion` with the following files:
+
+```json5
+// src/plugins/timelion/kibana.json
+{
+  "id": "timelion",
+  "server": true
+}
+```
+
+```ts
+// src/plugins/timelion/server/index.ts
+import { schema } from '@kbn/config-schema';
+import { PluginInitializerContext } from '../../core/server';
+import { TimelionPlugin } from './plugin';
+
+export const config = {
+  schema: schema.object({
+    enabled: schema.boolean({ defaultValue: true }),
+  });
+}
+
+export const plugin = (initContext: PluginInitializerContext) => new TimelionPlugin(initContext);
+
+export type TimelionConfig = typeof config.schema;
+export { TimelionSetup } from './plugin';
+```
+
+```ts
+// src/plugins/timelion/server/plugin.ts
+import { PluginInitializerContext, Plugin, CoreSetup } from '../../core/server';
+import { TimelionConfig } from '.';
+
+export class TimelionPlugin implements Plugin<TimelionSetup> {
+  constructor(private readonly initContext: PluginInitializerContext) {}
+
+  public setup(core: CoreSetup) {
+    return {
+      config$: this.initContext.config.create<TimelionConfig>(),
+    };
+  }
+
+  public setup() {}
+  public start() {}
+}
+
+export interface TimelionSetup {
+  config$: Observable<TimelionConfig>;
+}
+```
+
+With the New Platform plugin in place, you can then read this `config$`
+Observable from your legacy plugin:
+
+```ts
+import { take } from 'rxjs/operators';
+
+new kibana.Plugin({
+  async init(server) {
+    const { config$ } = server.newPlatform.setup.plugins.timelion;
+    const currentConfig = await config$.pipe(take(1)).toPromise();
+  }
+});
+```
+
+## HTTP Routes
+
+In the legacy platform, plugins have direct access to the Hapi `server` object
+which gives full access to all of Hapi's API. In the New Platform, plugins
+have access to the [HttpServiceSetup]() interface, which is exposed via the
+[CoreSetup] object injected into the `setup` method of server-side plugins.
+
+This interface has a different API with slightly different behaviors.
+- All input (body, query parameters, and URL parameters) must be validated using
+  the `@kbn/config-schema` package. If no validation schema is provided, these
+  values will be empty objects.
+- All exceptions thrown by handlers result in 500 errors. If you need a specific
+  HTTP error code, catch any exceptions in your handler and construct the
+  appropriate response using the provided response factory.
```

Comment:
Let's add a note that plugins **do not have to get rid of ** `Boom` completely. It's not a part of the platform public contract, but they may use it internally if they want. https://github.com/elastic/kibana/pull/47643#discussion_r333387385